data/tools/wesnoth/wmltools3.py

```diff
@@ -286,66 +294,126 @@ def isresource(filename):
     return ext and ext[1:] in resource_extensions
 
 def parse_macroref(start, line):
-    def handle_argument():
-        nonlocal opt_arg
-        nonlocal arg
-        nonlocal optional_args
-        nonlocal args
+    def handle_argument(buffer):
+        opt_arg = ""
+
+        arg = "".join(buffer)
+        if arg[0:1].isspace():
+            arg = arg[1:]
 
-        arg = arg.strip()
         # is this an optional argument?
         # argument names are usually made of uppercase letters, numbers and underscores
         # if they're optional, they're followed by an equal sign
         # stop matching on the first one, because the argument value might contain one too
         if re.match(r"^([A-Z0-9_]+?)=", arg):
             opt_arg, arg = arg.split("=", 1)
-        if arg.startswith('"') and arg.endswith('"'):
-            arg = arg[1:-1].strip()
         if opt_arg:
             optional_args[opt_arg] = arg
             opt_arg = ""
         else:
             args.append(arg)
-        arg = ""
+        buffer.clear()
+        return True
 
-    brackdepth = parendepth = 0
-    instring = False
-    optional_args = {}
     args = []
-    opt_arg = ""
-    arg = ""
+    optional_args = {}
+    buffer = []
+
+    depth = {
+        EQUALS: 0,
+        OPEN_BRACE: 0,
+        OPEN_PARENS: 0,
+        QUOTE: 0,
+    }
+    wrapper_stack = []
+    prev = ""
+
+    def close_token(token):
```

Comment:
Can you add a small comment to make clear at a quick glance when it's better to use `close_token()` or when `close_if_token()` should be preferred?

Reply:
Fair enough